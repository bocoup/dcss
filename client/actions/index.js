import {
    LOG_IN,
    LOG_OUT,
    SET_RUN,
    SET_SCENARIO,
    SET_SCENARIOS,
    SET_SLIDES,
    SET_USERS
} from './types';

export const logIn = username => ({
    type: LOG_IN,
    payload: {
        username,
        isLoggedIn: true
    }
});

export const logOut = username => ({
    type: LOG_OUT,
    payload: {
        username,
        isLoggedIn: false
    }
});

export const setRun = payload => ({
    type: SET_RUN,
    payload
});

export const setScenario = payload => ({
    type: SET_SCENARIO,
    payload
});

export const setScenarios = payload => ({
    type: SET_SCENARIOS,
    payload
});

export const setSlides = payload => ({
    type: SET_SLIDES,
    payload
});

export const setUsers = payload => ({
    type: SET_USERS,
    payload
});
