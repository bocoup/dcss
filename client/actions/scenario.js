import {
    GET_SCENARIO,
    GET_SCENARIO_SUCCESS,
    GET_SCENARIO_ERROR,
    GET_SCENARIOS,
    GET_SCENARIOS_SUCCESS,
    GET_SCENARIOS_ERROR,
    GET_SCENARIO_RUN_HISTORY,
    GET_SCENARIO_RUN_HISTORY_SUCCESS,
    GET_SCENARIO_RUN_HISTORY_ERROR,
    GET_SLIDES,
    GET_SLIDES_SUCCESS,
    GET_SLIDES_ERROR,
    SET_SCENARIO,
    // SET_SCENARIO_SUCCESS,
    // SET_SCENARIO_ERROR,
    SET_SCENARIOS,
    // SET_SCENARIOS_SUCCESS,
    // SET_SCENARIOS_ERROR,
    SET_SLIDES
} from './types';

export const getScenarios = () => async dispatch => {
    dispatch({ type: GET_SCENARIOS });
    try {
        const { scenarios, error } = await (await fetch(
            '/api/scenarios'
        )).json();
        if (error) {
            throw error;
        }

        dispatch({ type: GET_SCENARIOS_SUCCESS, scenarios });
        return scenarios;
    } catch (error) {
        const { message, status, stack } = error;
        dispatch({ type: GET_SCENARIOS_ERROR, status, message, stack });
    }
};

export const getScenario = id => async dispatch => {
    dispatch({ type: GET_SCENARIO, id });
    try {
        const { scenario, error } = await (await fetch(
            `/api/scenarios/${id}`
        )).json();

        if (error) {
            throw error;
        }
        dispatch({ type: GET_SCENARIO_SUCCESS, scenario });
        return scenario;
    } catch (error) {
        const { message, status, stack } = error;
        dispatch({ type: GET_SCENARIO_ERROR, status, message, stack });
    }
};

export const getScenarioRunHistory = params => async dispatch => {
    const { scenarioId, cohortId } = params;
    dispatch({ type: GET_SCENARIO_RUN_HISTORY });
    try {
        const endpoint = cohortId
            ? `/api/scenarios/${scenarioId}/cohort/${cohortId}/history`
            : `/api/scenarios/${scenarioId}/history`;

        const { history, error } = await (await fetch(endpoint)).json();

        if (error) {
            throw error;
        }

        dispatch({ type: GET_SCENARIO_RUN_HISTORY_SUCCESS, ...history });
        return { ...history };
    } catch (error) {
        const { message, stack, status } = error;
        dispatch({
            type: GET_SCENARIO_RUN_HISTORY_ERROR,
            message,
            stack,
            status
        });
        // pass along the error to the promise action
        throw error;
    }
};

export const getSlides = scenarioId => async dispatch => {
    dispatch({ type: GET_SLIDES, scenarioId });
    try {
        const { slides, error } = await (await fetch(
            `/api/scenarios/${scenarioId}/slides`
        )).json();

        if (error) {
            throw error;
        }
        dispatch({ type: GET_SLIDES_SUCCESS, slides });
        return slides;
    } catch (error) {
        const { message, status, stack } = error;
        dispatch({ type: GET_SLIDES_ERROR, status, message, stack });
    }
};

export const setScenario = scenario => ({
    type: SET_SCENARIO,
    scenario
});

export const setScenarios = scenarios => ({
    type: SET_SCENARIOS,
    scenarios
});

export const setSlides = slides => ({
    type: SET_SLIDES,
    slides
});
