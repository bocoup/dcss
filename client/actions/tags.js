import { GET_CATEGORIES_SUCCESS, GET_CATEGORIES_ERROR } from './types';

export const getCategories = () => async dispatch => {
  try {
    const res = await (await fetch('/api/tags/categories')).json();

    if (res.error) {
      throw res;
    }

    const {
      categories
    } = res;

    dispatch({ type: GET_CATEGORIES_SUCCESS, categories });
    return categories;
  } catch (error) {
    dispatch({ type: GET_CATEGORIES_ERROR, error });
    return null;
  }
};
