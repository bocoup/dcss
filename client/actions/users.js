import { GET_USERS_SUCCESS, GET_USERS_ERROR } from './types';

export const setUsers = users => ({
  type: GET_USERS_SUCCESS,
  users
});

export const getUsers = () => async dispatch => {
  try {
    const { users } = await (await fetch('/api/roles/all')).json();
    dispatch({ type: GET_USERS_SUCCESS, users });
    return users;
  } catch (error) {
    dispatch({ type: GET_USERS_ERROR, error });
    return error;
  }
};
