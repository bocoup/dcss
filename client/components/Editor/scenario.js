export function makeDefaultDescription({ title, description }) {
  let returnValue = description;
  if (!returnValue && title) {
    returnValue = `A scenario about "${title}"`;
  }
  return returnValue;
}
