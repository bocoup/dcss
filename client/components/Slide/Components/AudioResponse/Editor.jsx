import React, { Component } from 'react';
import PropTypes from 'prop-types';
import hash from 'object-hash';
import { Container, Form, Message } from '@components/UI';
import { type } from './meta';
import DataHeader from '@components/Slide/Components/DataHeader';
import ResponseRecall from '@components/Slide/Components/ResponseRecall/Editor';
import './AudioResponse.css';
import '@components/Slide/SlideEditor/SlideEditor.css';

class AudioResponseEditor extends Component {
  constructor(props) {
    super(props);
    const { header = '', prompt = '', recallId = '', responseId } = props.value;

    this.state = {
      header,
      prompt,
      recallId,
      responseId
    };
    this.onChange = this.onChange.bind(this);
    this.onRecallChange = this.onRecallChange.bind(this);
    this.updateState = this.updateState.bind(this);
    this.delayedUpdateState = this.delayedUpdateState.bind(this);
    this.timeout = null;
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);

    const { header, prompt, recallId, responseId } = this.props.value;

    const lastProps = {
      header,
      prompt,
      recallId,
      responseId
    };

    if (hash(this.state) !== hash(lastProps)) {
      this.updateState();
    }
  }

  delayedUpdateState() {
    if (this.timeout) {
      clearTimeout(this.timeout);
    }

    this.timeout = setTimeout(this.updateState, 500);
  }

  updateState() {
    const { header, prompt, recallId, responseId } = this.state;

    this.props.onChange({
      header,
      prompt,
      recallId,
      responseId,
      type
    });
  }

  onChange(event, { name, value }) {
    this.setState({ [name]: value }, this.delayedUpdateState);
  }

  onRecallChange({ recallId }) {
    this.setState({ recallId }, this.updateState);
  }

  render() {
    const { header, prompt, recallId } = this.state;
    const { scenarioId, slideIndex } = this.props;
    const { onChange, onRecallChange, updateState } = this;

    return (
      <Form>
        <Container fluid>
          <ResponseRecall
            isEmbedded={true}
            value={{ recallId }}
            scenarioId={scenarioId}
            slideIndex={slideIndex}
            onChange={onRecallChange}
          />

          <Form.TextArea
            label="Enter text content to display before the audio recording controls:"
            name="prompt"
            value={prompt}
            onChange={onChange}
            onBlur={updateState}
          />

          <DataHeader
            content={header}
            onChange={onChange}
            onBlur={updateState}
          />

          <Message
            icon="warning sign"
            color="orange"
            content="This component will fallback to a text input prompt when the participant's browser or device does not support audio recording."
          />
        </Container>
      </Form>
    );
  }
}

AudioResponseEditor.propTypes = {
  onChange: PropTypes.func.isRequired,
  scenarioId: PropTypes.any,
  slideIndex: PropTypes.any,
  value: PropTypes.shape({
    id: PropTypes.string,
    header: PropTypes.string,
    prompt: PropTypes.string,
    recallId: PropTypes.string,
    required: PropTypes.bool,
    responseId: PropTypes.string,
    type: PropTypes.oneOf([type])
  })
};

export default AudioResponseEditor;
