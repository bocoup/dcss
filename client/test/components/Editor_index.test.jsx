import React from 'react';
import assert from 'assert';
import {
  fetchImplementation,
  mounter,
  reduxer,
  snapshotter,
  state,
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';
import { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Editor from '../../components/Editor/index.jsx';

import {
  GET_CATEGORIES_SUCCESS,
  GET_SCENARIO_SUCCESS,
  GET_USERS_SUCCESS,
} from '../../actions/types';
import * as usersActions from '../../actions/users';
import * as scenarioActions from '../../actions/scenario';
import * as tagsActions from '../../actions/tags';
jest.mock('../../actions/users');
jest.mock('../../actions/scenario');
jest.mock('../../actions/tags');

const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

afterAll(() => {
  fetch.mockRestore();
});

beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);

  scenarioActions.copyScenario = jest.fn();
  scenarioActions.copyScenario.mockImplementation(() => async (dispatch) => {
    const scenario = {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    };
    dispatch({ type: GET_SCENARIO_SUCCESS, scenario });
    return scenario;
  });
  scenarioActions.deleteScenario = jest.fn();
  scenarioActions.deleteScenario.mockImplementation(() => async (dispatch) => {
    const scenario = {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    };
    dispatch({ type: GET_SCENARIO_SUCCESS, scenario });
    return scenario;
  });
  scenarioActions.getScenario = jest.fn();
  scenarioActions.getScenario.mockImplementation(() => async (dispatch) => {
    const scenario = {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    };
    dispatch({ type: GET_SCENARIO_SUCCESS, scenario });
    return scenario;
  });
  scenarioActions.setScenario = jest.fn();
  scenarioActions.setScenario.mockImplementation(() => async (dispatch) => {
    const scenario = {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    };
    dispatch({ type: GET_SCENARIO_SUCCESS, scenario });
    return scenario;
  });
  scenarioActions.endScenarioLock = jest.fn();
  scenarioActions.endScenarioLock.mockImplementation(() => async (dispatch) => {
    const scenario = {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    };
    dispatch({ type: GET_SCENARIO_SUCCESS, scenario });
    return scenario;
  });

  usersActions.getUsers = jest.fn();
  usersActions.getUsers.mockImplementation(() => async (dispatch) => {
    const users = [
      {
        id: 999,
        email: 'owner@email.com',
        username: 'owner',
        personalname: 'Owner Account',
        roles: ['owner'],
        is_owner: true,
        is_author: true,
        is_reviewer: false,
      },
    ];
    dispatch({ type: GET_USERS_SUCCESS, users });
    return users;
  });

  usersActions.getUsersByPermission = jest.fn();
  usersActions.getUsersByPermission.mockImplementation(
    () => async (dispatch) => {
      const users = [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ];
      dispatch({ type: GET_USERS_SUCCESS, users });
      return users;
    }
  );

  tagsActions.getCategories = jest.fn();
  tagsActions.getCategories.mockImplementation(() => async (dispatch) => {
    const categories = [];
    dispatch({ type: GET_CATEGORIES_SUCCESS, categories });
    return categories;
  });

  const statusOptions = [
    { id: 1, name: 'draft', description: 'Visible only to author' },
    { id: 2, name: 'public', description: 'Visible to everyone' },
    { id: 3, name: 'private', description: 'Visible only to logged in users' },
  ];
  fetchImplementation(fetch, 200, statusOptions);

  delete window.location;
  // eslint-disable-next-line
  window.location = {
    href: '',
  };

  commonProps = {
    history: {
      push() {},
    },
  };

  commonState = JSON.parse(JSON.stringify(original));
});

afterEach(() => {
  fetch.mockReset();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
});

test('Editor', () => {
  expect(Editor).toBeDefined();
});

test('Snapshot 1 1', async (done) => {
  const Component = Editor;

  const props = {
    ...commonProps,
    scenario: {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    },
    isCopyScenario: true,
    scenarioId: 1,
    scenarioUser: {
      id: 999,
      email: 'owner@email.com',
      username: 'owner',
      personalname: 'Owner Account',
      roles: ['owner'],
      is_owner: true,
      is_author: true,
      is_reviewer: false,
    },
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const wrapper = mounter(reduxed);
  expect(snapshotter(reduxed)).toMatchSnapshot();
  expect(snapshotter(wrapper)).toMatchSnapshot();

  const component = wrapper.findWhere((n) => {
    return n.type() === Component;
  });
  expect(snapshotter(component)).toMatchSnapshot();

  done();
});

test('Snapshot 2 1', async (done) => {
  const Component = Editor;

  const props = {
    ...commonProps,
    scenario: {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    },
    isNewScenario: true,
    scenarioId: 1,
    scenarioUser: {
      id: 999,
      email: 'owner@email.com',
      username: 'owner',
      personalname: 'Owner Account',
      roles: ['owner'],
      is_owner: true,
      is_author: true,
      is_reviewer: false,
    },
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const wrapper = mounter(reduxed);
  expect(snapshotter(reduxed)).toMatchSnapshot();
  expect(snapshotter(wrapper)).toMatchSnapshot();

  const component = wrapper.findWhere((n) => {
    return n.type() === Component;
  });
  expect(snapshotter(component)).toMatchSnapshot();

  done();
});

test('Snapshot 3 1', async (done) => {
  const Component = Editor;

  const props = {
    ...commonProps,
    scenario: {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    },
    scenarioId: 1,
    scenarioUser: {
      id: 999,
      email: 'owner@email.com',
      username: 'owner',
      personalname: 'Owner Account',
      roles: ['owner'],
      is_owner: true,
      is_author: true,
      is_reviewer: false,
    },
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const wrapper = mounter(reduxed);
  expect(snapshotter(reduxed)).toMatchSnapshot();
  expect(snapshotter(wrapper)).toMatchSnapshot();

  const component = wrapper.findWhere((n) => {
    return n.type() === Component;
  });
  expect(snapshotter(component)).toMatchSnapshot();

  done();
});

/*{INJECTION}*/

