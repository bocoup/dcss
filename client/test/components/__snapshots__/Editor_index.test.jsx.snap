// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1 1 1`] = `""`;

exports[`Snapshot 1 1 2`] = `""`;

exports[`Snapshot 1 1 3`] = `""`;

exports[`Snapshot 1 2 1`] = `""`;

exports[`Snapshot 2 1 1`] = `""`;

exports[`Snapshot 2 1 2`] = `""`;

exports[`Snapshot 2 1 3`] = `""`;

exports[`Snapshot 2 2 1`] = `""`;

exports[`Snapshot 3 1 1`] = `""`;

exports[`Snapshot 3 1 2`] = `""`;

exports[`Snapshot 3 1 3`] = `""`;

exports[`Snapshot 3 2 1`] = `""`;
